vimeo/stag-java
# Deterministic compilation

# Issue 
- https://github.com/vimeo/stag-java/issues/181

## Summary
Compilation was not deterministic due to usage of `HashSet` and `HashMap`. Switched to `LinkedHashMap` and `LinkedHashSet` in order to guarantee deterministic iteration and deterministic compilation. Added a functional test that validates consistent output between compilations. 

## How to Test
Let the tests run. See that the old test fails if the changes to the usage of the implementation of `Map` and `Set` are reverted.
